pennylane/templates/subroutines/temporary_and.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the TemporaryAnd template.
+"""
+
+from functools import lru_cache
+
+import pennylane as qml
+from pennylane.decomposition import add_decomps, register_resources
+from pennylane.operation import Operation
+from pennylane.wires import WiresLike
+
+
+class TemporaryAnd(Operation):
+    r"""TemporaryAnd(wires)
+
+    The TemporaryAnd operator is a three-qubit gate equivalent to an ``AND``, or reversible :class:`~pennylane.Toffoli`, gate that leverages extra information
+    about the target wire to enable more efficient circuit decompositions. The ``TemporaryAnd`` assumes the target qubit
+    to be initialized in :math:`|0\rangle`, while the ``Adjoint(TemporaryAnd)`` assumes the target output to be :math:`|0\rangle`.
+    For more details, see `Ryan Babbush et al.(2018), Fig 4 <https://arxiv.org/abs/1805.03662>`_.
+
+    .. note::
+
+        For correct usage of the operator, the user must ensure that the input or output is :math:`|0\rangle`
```

Comment:
Input and output? Or does it only matter in the independent context of computing vs un-computing? Regardless, we can probably be a bit more descriptive here.
```suggestion
        For correct usage of the operator, the user must ensure
        that the input before computation is :math:`|0\rangle`,
        and that the output after uncomputation is :math:`|0\rangle`.
```